### ISSUE TITLE
abstract methods in complex hierarchies with multiple inheritance are not always handled correctly

### ISSUE THREAD

**Issue by davidnich** [qorelanguage/qore#2741]
when a base class declares abstract methods and a child class inherits the base class and another class that implements the methods, then Qore recognizes that the methods are implemented and allows the class to be instantiated, but then cannot call them at runtime with an error as follows:
```
ABSTRACT-VARIANT-ERROR: cannot call abstract variant AbstractQorusCoreService::getOption(string field) directly"}
```
Either Qore should not allow this (i.e. such classes will be still marked as abstract and therefore cannot be instantiated) or it should be supported properly (i.e. concrete variants of abstract methods can be inherited from another class as originally intended)

**Comment by davidnich** [comment]
partially fixed, needs more attention

**Comment by davidnich** [comment]
fixed with #3070
